plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1565,6 +1569,7 @@ public void testOptimizeWithSortOrder()
     }
 
     @Test
+    @ResourceLock(value = "BaseIcebergConnectorTest", mode = READ_WRITE)
```

Comment:
@lukasz-stec Did you try the existing `AbstractTestQueryFramework#executeExclusively` method? 

@martint This is the 1st usage of `@ResourceLock` annotation. Can we use it for isolating tests?

Reply:
> @lukasz-stec Did you try the existing AbstractTestQueryFramework#executeExclusively method?

No, I did not. It is the first time I have seen it but it is also private and only works for a single query and not the entire test.

Reply:
It's a package-private and we already use the method for the entire test:
https://github.com/trinodb/trino/blob/b1d2302786dc98f509e07f340d75b2a0c41b7954/plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java#L2358-L2362

Reply:
Oh, right, I missed that. I think it should work in that case. If this is the preferred way to run a single test isolated, I can change that.
Please confirm.

Reply:
> I think it should work in that case. If this is the preferred way to run a single test isolated, I can change that.
Please confirm.

@ebyhr @martint Any feedback on this?